pkg/component/kubernetes/apiserver/configmaps.go

```diff
@@ -79,3 +89,19 @@ func (k *kubeAPIServer) reconcileConfigMapEgressSelector(ctx context.Context, co
 
 	return client.IgnoreAlreadyExists(k.client.Client().Create(ctx, configMap))
 }
+
+func (k *kubeAPIServer) reconcileConfigMapEnvoyConfig(ctx context.Context, configMap *corev1.ConfigMap) error {
+	if !k.values.VPN.Enabled && !k.values.VPN.HighAvailabilityEnabled {
```

Comment:
Is the condition correct? I would have assumed that just disabling VPN or HA VPN should be enough.

```suggestion
	if !k.values.VPN.Enabled || !k.values.VPN.HighAvailabilityEnabled {
```